jbmc/regression/jbmc/FloatToIntConversion/Test.java

```diff
@@ -0,0 +1,220 @@
+public class Test
+{
+  public static void floatToInt(int c, float a, float b)
+  {
+    switch(c)
+    {
+    case 0: 
+      if (a == 3.0f && b == 2.0f) {
+        int result = (int)(a / b);
+        assert result == 1;
+      }
+      break;
+    case 1: 
+      if (a == 3.0f && b == -2.0f) {
+        int result = (int)(a / b);
+        assert result == -1;
+      }
+      break;
+    case 2: 
+      if (a == 1.0f) {
+        // use constant here, -0.0f reverses cases 2 and 3.
+        int result = (int)(a / 0.0f);
+        // a/b = +inf
+        assert result == 2147483647;
```

Comment:
I'm going to trust you that this is the JVM semantics but ... I am surprised by this.

Reply:
I was also surprised until I ran it through the JVM.

Reply:
Cool.